test/test_nn.py

```diff
@@ -13324,6 +13324,27 @@ def test_MarginLoss_empty(self, device, dtype):
                 y = torch.ones(10, 0, device=device).type(torch.long)
                 mod(x, y)
 
+    @onlyOnCPUAndCUDA
+    @dtypes(torch.float, torch.double)
+    def test_adaptive_pooling_zero_batch(self, dtype):
```

Comment:
I think this should accept a `device` kwarg used below instead of creating & iterating over the `device_list`. This works locally for me, but let me know if you run into problems

Reply:
I used `@onlyOnCPUAndCUDA` and passed a device argument, but the code failed. You can see that other tests using adaptive pooling such as this one: https://github.com/pytorch/pytorch/blob/master/test/test_nn.py#L5161 use the same approach.